core-web/libs/portlets/edit-ema/portlet/src/lib/dot-ema-shell/dot-ema-shell.component.ts

```diff
@@ -104,59 +104,70 @@ export class DotEmaShellComponent implements OnInit, OnDestroy {
         seoProperties: DotPageToolUrlParams;
         error?: number;
     }> = this.store.shellProperties$.pipe(
-        map(({ currentUrl, page, host, languageId, siteId, error }) => ({
-            items: [
-                {
-                    icon: 'pi-file',
-                    label: 'editema.editor.navbar.content',
-                    href: 'content'
-                },
-                {
-                    icon: 'pi-table',
-                    label: 'editema.editor.navbar.layout',
-                    href: 'layout',
-                    isDisabled: !page.canEdit
-                },
-                {
-                    icon: 'pi-sliders-h',
-                    label: 'editema.editor.navbar.rules',
-                    href: `rules/${page.identifier}`,
-                    isDisabled: !page.canEdit
-                },
-                {
-                    iconURL: 'experiments',
-                    label: 'editema.editor.navbar.experiments',
-                    href: `experiments/${page.identifier}`
-                },
-                {
-                    icon: 'pi-th-large',
-                    label: 'editema.editor.navbar.page-tools',
-                    action: () => {
-                        this.pageTools.toggleDialog();
+        map(({ currentUrl, page, host, languageId, siteId, templateDrawed, error }) => {
+            const isLayoutDisabled = !page.canEdit || !templateDrawed;
+
+            if (isLayoutDisabled && this.router.url.includes('layout')) {
+                this.router.navigate(['./content'], { relativeTo: this.activatedRoute });
+            }
```

Comment:
Should we add this conditional to the `component` or at a `router` level?

Reply:
Think we need this in component
Because the user can be in the layout route and change the template to advanced template
In that case, a configuration in `router` level doesnt trigger, but we need to move the user to content route
(And we can for the shellProperties change)

Reply:
I don't know if the `includes` is bulletproof, looks like an user can break this by just creating a page with the word `layout` in it

Reply:
Check if you can get the exact current route with the `ActivatedRoute`